python-package/xgboost/sklearn.py

```diff
@@ -881,32 +853,20 @@ class XGBRegressor(XGBModel, XGBRegressorBase):
     __doc__ = "Implementation of the scikit-learn API for XGBoost regression.\n\n"\
         + '\n'.join(XGBModel.__doc__.split('\n')[2:])
 
+    def __init__(self, objective="reg:squarederror", **kwargs):
+        super().__init__(objective=objective, **kwargs)
+
 
 class XGBRFRegressor(XGBRegressor):
     # pylint: disable=missing-docstring
     __doc__ = "scikit-learn API for XGBoost random forest regression.\n\n"\
         + '\n'.join(XGBModel.__doc__.split('\n')[2:])
 
-    def __init__(self, max_depth=3, learning_rate=1, n_estimators=100,
-                 verbosity=1, objective="reg:squarederror", n_jobs=1,
-                 gpu_id=-1, gamma=0, min_child_weight=1,
-                 max_delta_step=0, subsample=0.8, colsample_bytree=1,
-                 colsample_bylevel=1, colsample_bynode=0.8, reg_alpha=0,
-                 reg_lambda=1e-5, scale_pos_weight=1, base_score=0.5,
-                 random_state=0, missing=None, **kwargs):
-        super(XGBRFRegressor, self).__init__(
-            max_depth=max_depth, learning_rate=learning_rate,
-            n_estimators=n_estimators, verbosity=verbosity,
-            objective=objective, booster='gbtree', n_jobs=n_jobs,
-            gpu_id=gpu_id, gamma=gamma, min_child_weight=min_child_weight,
-            max_delta_step=max_delta_step, subsample=subsample,
-            colsample_bytree=colsample_bytree,
-            colsample_bylevel=colsample_bylevel,
-            colsample_bynode=colsample_bynode,
-            reg_alpha=reg_alpha, reg_lambda=reg_lambda,
-            scale_pos_weight=scale_pos_weight,
-            base_score=base_score, random_state=random_state, missing=missing,
-            **kwargs)
+    def __init__(self, learning_rate=1, subsample=0.8, colsample_bynode=0.8,
+                 reg_lambda=1e-5, **kwargs):
+        super().__init__(learning_rate=learning_rate, subsample=subsample,
```

Comment:
Some parameters doesn't make sense for RF wrapper.  For example, the `n_estimators` means number of trees in random forest, while for other regressor `n_estimators` means number of boosting rounds.

Reply:
I'm not sure, but I think this behavior also occurs in original code. Maybe it is better to remove `n_estimators` from `XGBModel` and add it only to classifiers, but for regressors create separate parameter `n_boosting_rounds`?